### PR TITLE
docs(backup/s3): add readme content

### LINKED ISSUE #10219: S3 backup store documentation

### PR DESCRIPTION
**Pull request by lenaschoenburg**
closes #10219

### CODE REVIEW

File: backup-stores/s3/README.md [hunk 1]
```diff
@@ -1,7 +1,59 @@
...
+## Internals
+
+A backup consists of 2 objects used for managing the backup and the objects
+containing the backup data itself.
+All object keys are prefixed by `partitionId/checkpointId/nodeId` which uniquely identifies a
+backup:
+
+* `metadata.json`: A _metadata_ object, containing metadata serialized as JSON
+* `status.json`: A _status_ object, containing the current status serialized as JSON
+* `snapshots/*` Objects for _snapshot files_, additionally prefixed with 'snapshot'
```
Comment (deepthidevaki) [inline backup-stores/s3/README.md:57]:
```suggestion
* `snapshots/*` Objects for _snapshot files_'
```

File: backup-stores/s3/README.md [hunk 2]
```diff
@@ -1,7 +1,59 @@
...
+* `segments/*` Objects for _segment files_, additionally prefixed with 'segments'
```
Comment (deepthidevaki) [inline backup-stores/s3/README.md:58]:
```suggestion
* `segments/*` Objects for _segment files_
```

File: backup-stores/s3/README.md [hunk 3]
```diff
@@ -1,7 +1,59 @@
...
+* There is no concept of backup rotation so backups will accumulate unless manually deleted.
+* Individual files in a backup may not exceed 5GiB, otherwise the upload may fail.
 * If the backup consists of more than 1000 files, deleting a backup may leave behind some objects.
 
+## Internals
+
+A backup consists of 2 objects used for managing the backup and the objects
+containing the backup data itself.
+All object keys are prefixed by `partitionId/checkpointId/nodeId` which uniquely identifies a
+backup:
```
Comment (deepthidevaki) [inline backup-stores/s3/README.md:53]:
```suggestion
A backup consists of following objects prefixed by `partitionId/checkpointId/nodeId/` :
```

**Review by deepthidevaki — APPROVED** [review]
Great! Please check my comments before merging.

**Comment by oleschoenburg** [comment]
Thanks @deepthidevaki, I liked your suggestions :+1: 

bors r+

**Pull request merged**